python/ray/train/torch/train_loop_utils.py

```diff
@@ -467,7 +467,11 @@ def get_device(self) -> torch.device:
         if torch.cuda.is_available():
             gpu_ids = ray.get_gpu_ids()
             if len(gpu_ids) > 0:
-                device_id = gpu_ids[0]
+                gpu_id = gpu_ids[0]
+                cuda_visible_list = list(
+                    map(int, ray._private.utils.get_cuda_visible_devices())
```

Comment:
Can we manually get the cuda visible devices via the env var instead of using a private ray API?

Or move the private api to `ray.util` and mark it as a developer api.